db/seeders/donations_seeder.rb

```diff
@@ -0,0 +1,78 @@
+class DonationsSeeder
+  attr_accessor :organization
+
+  def self.seed(organization)
+    new(organization).seed
+  end
+
+  def initialize(organization)
+    @organization = organization
+  end
+
+  def seed
+    20.times.each do
+      donation = donate(source)
+
+      rand(1..5).times.each { create_line_item(donation) }
+
+      donation.reload
+      donation.storage_location.increase_inventory(donation)
+    end
+  end
+
+  private
+
+  def source
+    Donation::SOURCES.values.sample
+  end
+
+  def donate(source)
+    case source
+    when Donation::SOURCES[:diaper_drive]
+      create_drive_participant(source)
+    when Donation::SOURCES[:donation_site]
+      create_dontation_site(source)
+    when Donation::SOURCES[:manufacturer]
+      create_manufacter(source)
+    else
+      donation_create(source, {})
+    end
+  end
+
+  def create_drive_participant(source)
+    donation_create(
+      source, diaper_drive_participant: random_record_for_org(organization, DiaperDriveParticipant)
+    )
+  end
+
+  def create_dontation_site(source)
+    donation_create(
+      source, donation_site: random_record_for_org(organization, DonationSite)
+    )
+  end
+
+  def create_manufacter(source)
+    donation_create(
+      source, manufacturer: random_record_for_org(organization, Manufacturer)
+    )
+  end
```

Comment:
I recommend renaming these methods to include 'with' or something like it, and including 'donation' in the name, e.g. `create_donation_with_manufacturer`.

Also, there is a typo, an extra 't', in `create_dontation_site`.

Reply:
Nice.